File: lib/nitro/QuantumUI.ts
// Web / non-native fallback — pure JS implementation.
// Metro resolves QuantumUI.native.ts on iOS/Android; this file is used on web.
export function multiply(a: number, b: number): number {
  return a * b;
}

// Web / non-native fallback — no SF Symbols off iOS; callers fall back to an empty icon box.
export function renderSFSymbol(
  _name: string,
  _size: number,
  _color: number,
  _weight: string,
  _multicolor: boolean,
): string {
  return '';
}

File: android/src/main/java/com/margelo/nitro/com/vritti/quantumui/QuantumUI.kt
package com.margelo.nitro.com.vritti.quantumui

import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class QuantumUI : HybridQuantumUISpec() {
  override fun multiply(a: Double, b: Double): Double {
    return a * b
  }

  // SF Symbols are iOS-only; DynamicIcon.android.tsx renders Material symbols instead,
  // so this is never called at runtime. Spec contract: '' on non-iOS platforms.
  override fun renderSFSymbol(
    name: String,
    size: Double,
    color: Double,
    weight: String,
    multicolor: Boolean,
  ): String = ""
}
